temperature/utils.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+import httpx
+from fastapi import Depends
+from sqlalchemy.ext.asyncio import AsyncSession
+from dotenv import load_dotenv
+
+from weather.crud import get_all_cities
+from dependencies import get_db
+
+load_dotenv()
+
+API_URL = os.getenv("API_URL")
+API_KEY = os.getenv("API_KEY")
+
+
+async def check_service_connection():
+    async with httpx.AsyncClient() as client:
+        response = await client.get(
+            f"{API_URL}/current.json", params={"key": API_KEY, "q": "London"}
+        )
+    if response.status_code != 200:
+        raise HTTPException(status_code=response.status_code, detail=response)
```

Comment:
The `HTTPException` is raised with `detail=response`, which will include the entire response object. Consider using `response.text` or `response.json()` to provide a more informative error message.